src/jit/flowgraph.cpp

```diff
@@ -17495,6 +17499,12 @@ BasicBlock*         Compiler::fgAddCodeRef(BasicBlock*      srcBlk,
                             break;
 #endif // COR_JIT_EE_VERSION
 
+    // SCK_NONE can appear on range checks that optimization proved will succeed
+    // but which it failed to remove for mechanical reasons; use failfast for
+    // these cases.
+    case SCK_NONE:          helper = CORINFO_HELP_FAIL_FAST;
```

Comment:
What would you prefer?  The check has been proven in-bounds, this is unreachable code being emitted.